src/apprunner/explorer/apprunnerServiceNode.ts

```diff
@@ -0,0 +1,173 @@
+        private info: AppRunner.Service,
+        private currentOperation: AppRunner.OperationSummary & { Type?: ServiceOperation } = {}
+    ) {
+        super(
+            'App Runner Service',
+            parent.region,
+            localize('AWS.explorerNode.apprunner.nologs', '[No App Runner logs found]')
+        )
+        this.iconPath = {
+            dark: vscode.Uri.file(ext.iconPaths.dark.apprunner),
+            light: vscode.Uri.file(ext.iconPaths.light.apprunner),
+        }
+        this.id = `AppRunnerService-${info.ServiceName}`
+        this.name = info.ServiceName
+        this.arn = info.ServiceArn
+
+        this.update(info)
+    }
+
+    public getInfo(): AppRunner.Service {
+        return this.info
+    }
+
+    public getUrl(): string {
+        return `https://${this.info.ServiceUrl}`
+    }
+
+    protected async getLogGroups(client: CloudWatchLogsClient): Promise<Map<string, CloudWatchLogs.LogGroup>> {
+        return toMap(
+            await toArrayAsync(
+                client.describeLogGroups({
+                    logGroupNamePrefix: `/aws/apprunner/${this.info.ServiceName}/${this.info.ServiceId}`,
+                })
+            ),
+            configuration => configuration.logGroupName
+        )
+    }
+
+    private setLabel(): void {
+        const displayStatus = this.currentOperation.Type
+            ? OPERATION_STATUS[this.currentOperation.Type]
+            : `${this.info.Status.charAt(0)}${this.info.Status.slice(1).toLowerCase().replace(/\_/g, ' ')}`
+        this.label = `${this.info.ServiceName} [${displayStatus}]`
+    }
+
+    public update(info: AppRunner.ServiceSummary | AppRunner.Service): void {
+        // update can be called multiple times during an event loop
+        // this would rarely cause the node's status to appear as 'Operation in progress'
+        this.lock.acquire(this.info.ServiceId, done => {
+            const lastLabel = this.label
+            this.updateInfo(info)
+            this.updateStatus(lastLabel)
+            done()
+        })
+    }
+
+    private updateStatus(lastLabel?: string): void {
+        if (this.label !== lastLabel && lastLabel !== 'App Runner Service') {
+            this.refresh()
+        }
+
+        if (this.info.Status === 'DELETED') {
+            this.parent.deleteNode(this.info.ServiceArn)
+            this.parent.refresh()
+        }
+
+        if (this.info.Status === 'OPERATION_IN_PROGRESS') {
+            this.parent.startPolling(this.info.ServiceArn)
+        } else if (this.currentOperation.Type !== undefined) {
+            this.currentOperation.Id = undefined
+            this.currentOperation.Type = undefined
+            this.setLabel()
+            this.parent.stopPolling(this.info.ServiceArn)
+        }
+    }
+
+    private async updateOperation(): Promise<void> {
+        const operations = (await this.client.listOperations({ MaxResults: 1, ServiceArn: this.info.ServiceArn }))
```

Comment:
try/catch this? Goes for all other network calls as well.

Reply:
Currently it's handled in a different PR, though I'm thinking it would be wise to move that logic here since some of these functions are public.